### PR TITLE
fix(up): improve local content repository development

### LINKED ISSUE #714: improve local content repository development
it should be possible to quickly render content from the local git repository, if it matches the current origin. 

for example:

```
strains:
  - name: default
    content:  https://github.com/adobe/developer.adobe.com.git#master
    code:  https://github.com/adobe/developer.adobe.com.git#master
    static:  https://github.com/adobe/developer.adobe.com.git/htdocs#master
```
When running `hlx up` in a local checkout of `https://github.com/adobe/developer.adobe.com.git`, the content and static content should be served from the local repo instead from github.

Also, it should be possible to disable this feature, or even specify more repositories that should be emulated.

### Suggest

`hlx up` will have a new flag `--git` or `--git-serve` or `--mount` or `--repo` which takes **1** directory argument that must point to a local checkout of a content repository. `hlx up` will then start a git-server on those directory and map all repository references in the strains.

### PR DESCRIPTION
**Pull request by tripodsan**
fix #714 

needs https://github.com/adobe/helix-simulator/pull/178

### CODE REVIEW

**Comment by tripodsan** [comment]
@kptdobe any objection to merge this?

File: README.md [hunk 1]
```diff
@@ -221,6 +223,119 @@ In the example above, there are three strains: `default` serves content from `ww
...
+will never be selected, because the `localhost:3000` host header will not match the specified `url` 
+condition.
+
+Starting the server with:  
+
+```
+$ hlx up --host=www.project-helix.io
+```
+
+Solves the latter problem. the `--host` argument internally overrides the `request.header`, so that
```
Comment (filmaj) [inline README.md]: I feel like the `--host` flag should be described in its own dedicated and separate section.

File: README.md [hunk 2]
```diff
@@ -221,6 +223,119 @@ In the example above, there are three strains: `default` serves content from `ww
...
+│   ├── helix-config.yaml
+│   └── index.md
+└── welcome-api/
+    └── index.md
+```
+
+We can now launch the server with the respective `--local-repo` arguments: 
+
+```
+$ hlx up --host=www.project-helix.io --local-repo=. --local-repo=../welcome-api
```
Comment (filmaj) [inline README.md]: There was a suggestion to have the argument to the `--local-repo` be optional (https://github.com/adobe/helix-cli/issues/714#issuecomment-480174470) - if that is the case, let's update the example here.

File: README.md [hunk 3]
```diff
@@ -221,6 +223,119 @@ In the example above, there are three strains: `default` serves content from `ww
...
+
+Now the server will transiently reconfigure the strains, so that the emulated repositories are used.
+
+> **Note**: If you turn on `--log-level=debug` you should see log entries for the emulated repositories:   
+```
+[hlx] debug: git emulating https://github.com/helix/welcome.git via http://127.0.0.1:52270/helix/github.com--helix--welcome.git#master from './'
+[hlx] debug: git emulating https://github.com/helix/welcome-api.git via http://127.0.0.1:52270/helix/github.com--helix-welcome-api.git#master from '../welcome-api'
+```
+
+For convenience, you can also specify the arguments in an `.env` file:
```
Comment (filmaj) [inline README.md]: Can we explicitly document how to configure the `HLX_LOCAL_REPO` env var? It seems like it must be comma-separated for multiple repos. If that is so, can we document this somewhere here? Perhaps as a comment in the `dotenv` example?

File: README.md (hunk 1 above)
Comment (tripodsan) [reply README.md]: I just want to document the bare minimum. I think all this should go somewhere else, and not in the README of this project. but I agree.

File: README.md (hunk 2 above)
Comment (tripodsan) [reply README.md]: yes. but this is explained above (~ line 260). I thought that when specifying 2 repositories, it is more clear to explicitly list all.

File: README.md (hunk 1 above)
Comment (filmaj) [reply README.md]: OK, makes sense.

File: README.md (hunk 3 above)
Comment (tripodsan) [reply README.md]: see #698

**Pull request merged**